recipes/gdal/post_3.5.0/patches/3.5.1/0-replace-find-package.patch

```diff
@@ -0,0 +1,5569 @@
+diff --git a/alg/CMakeLists.txt b/alg/CMakeLists.txt
```

Comment:
this patch file is too big for maintenance. Could you try to simplify it?

Reply:
So let's see what I changed there:

1. Replace target names Blosc::Blosc -> c-blosc::c-blosc. I can use cmake alias for that, I believe.
2. Remove FindXXX.cmake from cmake/modules/packages. Instead, I removed the folder from CMAKE_MODULE_PATH.
3. replace gdal_target_link_libraries(PRIVATE) with target_link_libraries(PUBLIC) well that should go upstream, so maybe once we won't need it anymore. In the mean time, let's keep it.
4. Replace DefineFindPackage2 with ConanFindPackage.cmake. Since this is just swapping cmake functions, it should be easy to port to next versions. This allows to treat many dependencies the same way and makes sure we do not link with system-installed libs.
5. remove pkg-config gdal.pc generation. This is causing problems and we do not want that file anyway.

The patch is now much smaller.